codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -238,82 +172,26 @@ public T getAndRemove(T name, T defaultValue) {
     @Override
     public List<T> getAll(T name) {
         checkNotNull(name, "name");
-        List<T> values = new ArrayList<T>(4);
-        int h = hashCodeGenerator.generateHashCode(name);
+
+        LinkedList<T> values = new LinkedList<T>();
```

Comment:
what about checking if the headers are empty first and if so return an emptyList ?